src/Stryker.Core/Stryker.Core/ProjectComponents/TestProjects/TestProject.cs

```diff
@@ -1,14 +1,66 @@
 using System;
 using System.Collections.Generic;
+using System.IO.Abstractions;
 using System.Linq;
+using System.Text;
 using Buildalyzer;
+using Microsoft.CodeAnalysis;
+using Microsoft.CodeAnalysis.CSharp;
+using Stryker.Core.Exceptions;
+using Stryker.Core.MutantFilters.Extensions;
 
 namespace Stryker.Core.ProjectComponents.TestProjects
 {
     public sealed class TestProject : IEquatable<TestProject>
     {
-        public IAnalyzerResult TestProjectAnalyzerResult { get; init; }
-        public IEnumerable<TestFile> TestFiles { get; init; } = new List<TestFile>();
+        private readonly IFileSystem _fileSystem;
+
+        public IAnalyzerResult TestProjectAnalyzerResult { get; }
+
+        public IEnumerable<TestFile> TestFiles { get; }
+
+        public TestProject(IFileSystem fileSystem, IAnalyzerResult testProjectAnalyzerResult)
+        {
+            ValidateIsTestProject(testProjectAnalyzerResult);
+
+            _fileSystem = fileSystem ?? new FileSystem();
+
+            TestProjectAnalyzerResult = testProjectAnalyzerResult;
+
+            var testFiles = new List<TestFile>();
+            foreach (var file in testProjectAnalyzerResult.SourceFiles)
+            {
+                var source = _fileSystem.File.ReadAllText(file);
+                if (IsNotGenerated(file, source))
```

Comment:
Does it filter some "generated tests"? What this can be? And why are they filtered out then?

Reply:
I am filtering out generated files such as AssemblyInfo.cs so that they do not show up in the testreport, as we do for source projects as well.